server/src/main/java/org/eclipse/lsp/cobol/core/model/tree/Node.java

```diff
@@ -15,30 +15,34 @@
 package org.eclipse.lsp.cobol.core.model.tree;
 
 import com.google.common.collect.ImmutableList;
+import org.eclipse.lsp.cobol.core.model.Locality;
 import org.eclipse.lsp.cobol.core.model.SyntaxError;
-import org.eclipse.lsp4j.Location;
 
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
 import java.util.stream.Stream;
 
 /**
  * The class represents a Node in source structure tree.
  */
 public abstract class Node {
-  private final Location location;
+  private static final AtomicLong ID_COUNTER = new AtomicLong();
```

Comment:
👍

Reply:
Yeah, I plan to use it in serialization or keep track of the three nodes we visited.